docs/eternalcore/using-API.md

```diff
@@ -0,0 +1,105 @@
+# EternalCore Developer API
+
+As part of open-source community, we support developers who want to create their own plugins using our API. We provide a simple and easy to use API that allows you to create your own plugins using our API.
+
+## 📚 Dependency Management
+
+To use our work in your plugin, You need to install correct artifact for your project. Our plugin supports Maven, Gradle Kotlin, Gradle Groovy and Gradle Groovy and SBT.
+To use latest release check [maven repository](https://repo.eternalcode.pl/#/releases/com/eternalcode/eternalcore-api).
+
+### Add repository:
+
+For Gradle:
+```groovy
+maven {
+    url = uri("https://repo.eternalcode.pl/releases")
+}
+```
+
+For Maven:
+```xml
+<repository>
+    <id>eternalcode-reposilite-releases</id>
+    <name>EternalCode Repository</name>
+    <url>https://repo.eternalcode.pl/releases</url>
+</repository>
+```
+
+### Add dependency:
+
+For Gradle:
+```groovy
+compileOnly("com.eternalcode:eternalcore-api:1.1.0")
+```
+
+For Maven:
+```xml
+<dependency>
+    <groupId>com.eternalcode</groupId>
+    <artifactId>eternalcore-api</artifactId>
+    <version>1.1.0</version>
+    <scope>provided</scope>
+</dependency>
+```
+
+You must also add dependency inside `plugin.yml` file, this is required to load our plugin before your plugin, so they can access our API.
+```yaml
+depend: [EternalCore]
+```
```

Comment:
The PaperMC introduced the Paper Plugin System, which is a plugin alternative to the spigot plugin system, I suggest adding code group with groups: `plugin.yml, paper-plugin.yml` and see how to declare depend in paper-plugin.yml: https://docs.papermc.io/paper/dev/getting-started/paper-plugins#dependency-declaration


```yaml
dependencies:
  server:
    EternalCore:
      load: OMIT
      required: true
      join-classpath: true
```